### PR TITLE
add few widgets + expand ak functionality

### PR DESCRIPTION
**Pull request by oshtaier**

### CODE REVIEW

File: airgun/widgets.py [hunk 1]
```diff
@@ -267,3 +274,136 @@ def fill(self, values):
...
+            <div class="modal-footer ng-scope">
+                <button class="btn btn-danger" ng-click="ok()">
+                <button class="btn ..." ng-click="cancel()"...>
+
+    Locator example::
+
+        //div[@class='modal-content']
+
+    """
+    ROOT = "//div[@class='modal-content']"
```
Comment (abalakh) [inline airgun/widgets.py]: ".//div[...]"

Comment (abalakh) [reply airgun/widgets.py]: Btw, why `ConfirmationDialog` is a widget, it sounds like a view to me?

File: airgun/widgets.py [hunk 2]
```diff
@@ -267,3 +274,136 @@ def fill(self, values):
...
+            <li class="path-list-item ng-scope"...>
+                <label class="path-list-item-label...>
+                    <input type="checkbox"...>
+
+    Locator example::
+
+        //ul[@class='path-list']
+
+    """
+    ROOT = "//ul[@class='path-list']"
```
Comment (abalakh) [inline airgun/widgets.py]: ".//ul[...]"

File: airgun/widgets.py [hunk 3]
```diff
@@ -267,3 +274,136 @@ def fill(self, values):
...
+        """Identify whether specific checkbox is selected or not"""
+        return 'ng-not-empty' in self.browser.get_attribute('class', locator)
+
+    def select(self, locator, value):
+        """Select or deselect checkbox depends on the value passed"""
+        value = bool(value)
+        current_value = self.cb_selected(locator)
+        if value == current_value:
+            return False
+        else:
```
Comment (abalakh) [inline airgun/widgets.py]: Pls drop this else, it's not required after `return` in previous condition

File: airgun/widgets.py [hunk 4]
```diff
@@ -267,3 +274,136 @@ def fill(self, values):
...
+        //ul[@class='path-list']
+
+    """
+    ROOT = "//ul[@class='path-list']"
+    LABELS = "./li/label[contains(@class, path-list-item-label)]"
+    CHECKBOX = (
+        ".//input[@ng-model='item.selected'][parent::label[contains(., '{}')]]"
+    )
+
+    def cb_selected(self, locator):
```
Comment (abalakh) [inline airgun/widgets.py]: s/cb_selected/checkbox_selected/

File: airgun/widgets.py [hunk 5]
```diff
@@ -267,3 +274,136 @@ def fill(self, values):
...
+        return checkboxes
+
+    def fill(self, values):
+        """Assign value for specific checkbox from group
+
+        :param values: dictionary that consist of checkbox name and value
+            that should be assigned to that checkbox
+        """
+        checkbox_locator = self.CHECKBOX.format(values.get('entity_name'))
+        return self.select(checkbox_locator, values.get('select'))
```
Comment (abalakh) [inline airgun/widgets.py]: Helpers like `select` and `cb_selected` accept locators and create web elements multiple time. Are you sure that's the optimal way? I'd rather form web element for checkbox here and passed that single element to all the helpers.

File: airgun/widgets.py [hunk 6]
```diff
@@ -267,3 +274,136 @@ def fill(self, values):
...
+            checkboxes.append({'name': name, 'value': value})
+        return checkboxes
+
+    def fill(self, values):
+        """Assign value for specific checkbox from group
+
+        :param values: dictionary that consist of checkbox name and value
+            that should be assigned to that checkbox
+        """
+        checkbox_locator = self.CHECKBOX.format(values.get('entity_name'))
```
Comment (abalakh) [inline airgun/widgets.py]: as per discussion, let's drop "entity_name" and "value" keys and just pass name as a key, e.g. 'foo': True

File: airgun/views/activationkey.py [hunk 7]
```diff
@@ -14,13 +19,28 @@ def is_displayed(self):
...
     @property
     def is_displayed(self):
         return self.browser.wait_for_element(
             self.name, exception=False) is not None
+
+
+class ActivationKeyEditView(BaseLoggedInView):
+    return_to_all = Text("//a[text()='Activation Keys']")
+    action_list = SelectActionList()
+    dialog = ConfirmationDialog()
```
Comment (abalakh) [inline airgun/views/activationkey.py]: Have you checked cfme code? Separate dialog window as a widget doesn't sound right to me, i'd rather expect it to be a view, which appears during this view's `fill`. Does it make sense?

File: airgun/widgets.py (hunk 5 above)
Comment (oshtaier) [reply airgun/widgets.py]: You can't do that as `fill()` and `read()` has no connection with each other

File: airgun/widgets.py (hunk 6 above)
Comment (oshtaier) [reply airgun/widgets.py]: as per discussion - we need to use:
`mydictionary.keys()[0]` and `mydictionary[mydictionary.keys()[0]]`
that is not the good idea.
Any alternatives without loop?

File: airgun/views/activationkey.py (hunk 7 above)
Comment (oshtaier) [reply airgun/views/activationkey.py]: Yes, I am sure. It is simple control with two buttons

**Pull request merged**